gpu4pyscf/hessian/rks.py

```diff
@@ -50,19 +50,38 @@ def partial_hess_elec(hessobj, mo_energy=None, mo_coeff=None, mo_occ=None,
 
     if mf.do_nlc():
         raise NotImplementedError
+
     omega, alpha, hyb = ni.rsh_and_hybrid_coeff(mf.xc, spin=mol.spin)
     with_k = ni.libxc.is_hybrid_xc(mf.xc)
-    de2, ej, ek = rhf_hess._partial_hess_ejk(hessobj, mo_energy, mo_coeff, mo_occ,
-                                             atmlst, max_memory, verbose,
-                                             with_k=with_k)
-    de2 += ej  # (A,B,dR_A,dR_B)
+    j_factor = 1.
+    k_factor = 0.
     if with_k:
-        de2 -= hyb * ek
-    if abs(omega) > 1e-10 and abs(alpha-hyb) > 1e-10:
+        if omega == 0:
+            k_factor = hyb
+        elif alpha == 0: # LR=0, only SR exchange
+            pass
+        elif hyb == 0: # SR=0, only LR exchange
+            k_factor = alpha
+        else: # SR and LR exchange with different ratios
+            k_factor = alpha
+    de2, ejk = rhf_hess._partial_hess_ejk(hessobj, mo_energy, mo_coeff, mo_occ,
+                                          atmlst, max_memory, verbose,
+                                          j_factor, k_factor)
+    de2 += ejk  # (A,B,dR_A,dR_B)
+    if with_k and omega != 0:
+        j_factor = 0.
+        omega = -omega # Prefer computing the SR part
+        if alpha == 0: # LR=0, only SR exchange
+            k_factor = hyb
+        elif hyb == 0: # SR=0, only LR exchange
+            # full range exchange was computed in the previous step
+            k_factor = -alpha
+        else: # SR and LR exchange with different ratios
+            k_factor = hyb - alpha # =beta
```

Comment:
k_factor is simply `hyb - alpha`. The comments still explain the different cases.

Reply:
This part computes SR only.